### PR TITLE
Go vunlncheck fix

### PR DESCRIPTION
**Pull request by nckturner**
**What this PR does / why we need it**:
* Replaces https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/638
* Go version bump, fixes govulncheck test failure

### CODE REVIEW

File: cmd/aws-iam-authenticator/server.go [hunk 1]
```diff
@@ -52,15 +53,16 @@ var serverCmd = &cobra.Command{
 		var err error
 		fmt.Printf("Authenticator Version: %q, %q\n", pkg.Version, pkg.CommitID)
 		metrics.InitMetrics(prometheus.DefaultRegisterer)
-		stopCh := signals.SetupSignalHandler()
+		cmd.Context()
```
Comment (nckturner) [inline cmd/aws-iam-authenticator/server.go:56]: @micahhausler what are we doing with this?  https://github.com/spf13/cobra/blob/main/command.go#L257-L266

File: pkg/mapper/configmap/configmap.go [hunk 2]
```diff
@@ -50,16 +50,16 @@ func New(masterURL, kubeConfig string) (*MapStore, error) {
...
 		for {
 			select {
-			case <-stopCh:
-				logrus.Info("stopCh is closed in startLoadConfigMap")
+			case <-ctx.Done():
+				logrus.Info("Context is complete in startLoadConfigMap")
 				return
 			default:
-				watcher, err := ms.configMap.Watch(context.TODO(), metav1.ListOptions{
+				watcher, err := ms.configMap.Watch(context.Background(), metav1.ListOptions{
```
Comment (nckturner) [inline pkg/mapper/configmap/configmap.go:62]: @micahhausler originally you had this set to the same context that was passed in to startLoadConfigMap(), but this was causing the context to be closed when the watch call failed.  I'm not sure what the right thing to do here is though.

File: cmd/aws-iam-authenticator/server.go (hunk 1 above)
Comment (micahhausler) [reply cmd/aws-iam-authenticator/server.go:56]: I think we can drop it

File: pkg/mapper/configmap/configmap.go (hunk 2 above)
Comment (micahhausler) [reply pkg/mapper/configmap/configmap.go:62]: ah ok good catch

**Comment by nckturner** [comment]
Need this PR for integration tests: https://github.com/kubernetes/test-infra/pull/31295

**Pull request closed without merging**